airflow/models/dagrun.py

```diff
@@ -54,25 +54,27 @@ class DagRun(Base, LoggingMixin):
     _state = Column('state', String(50), default=State.RUNNING)
     run_id = Column(String(ID_LEN))
     external_trigger = Column(Boolean, default=True)
+    run_type = Column(String(50), nullable=True)
     conf = Column(PickleType)
 
     dag = None
 
     __table_args__ = (
-        Index('dag_id_state', dag_id, _state),
+        Index('dag_id_state_type', dag_id, _state, run_type),
```

Comment:
```suggestion
        Index('dag_id_state', dag_id, _state),
        Index('dag_run_type', run_type),
```

OR

```
        Index('dag_id_state', dag_id, _state),
        Index('dag_id_run_type', dag_id, run_type),
```

Maybe 2 indexes as most of the times we are filtering based on DagRunType and DagId 

WDYT? @turbaszek @ashb @mik-laj

Reply:
When do we filter by Dag id and run type?

Reply:
https://github.com/apache/airflow/blob/738667082d32d3ef93ec2cd6c3735ff3691ba1cc/airflow/jobs/scheduler_job.py#L567-L575

Reply:
```
Index('dag_run_type', run_type),
```

This alone should help

Reply:
@kaxil do you suggest single index `Index('dag_run_type', run_type)`? I must admit that I was running some tests with and without `Index('dag_id_state_type', dag_id, _state, run_type)` but there was no big difference